const testdata = require("../fixtures/csvjson.json");

describe('Test the app', function() {
   Object.keys(testdata).forEach(function(page, i){
      it(`Test Case For: ${page}`, function() {
	   let testDataRow = testdata[page];
	   let data = {
          name: testDataRow.name,
          email: testDataRow.email,
	      mobilenumber: testDataRow.mobilenumber
         };
        
		//cy.log(data.name)
		cy.on('uncaught:exception', (err, runnable) => {
    expect(err.message).to.include('something about the error')
    // using mocha's async done callback to finish
    // this test so we prove that an uncaught exception
    // was thrown
    done()
    // return false to prevent the error from form-control
    // failing this test
    return false
  })
        cy.wait(1000)
		cy.visit('https://nbtfeminastar.com/bio/?id=1027')
	    //cy.wait(3000);
		cy.get('.social-share > .btn').click({force: true})
		cy.get('form').within(($form) => {
  // you have access to the found form via
  // the jQuery object $form if you need it

  // cy.get() will only search for elements within form,
  // not within the entire document
  cy.wait(1000)
  cy.get('input[name="name"]').type(data.name)
  cy.wait(5000)
  cy.get('input[name="email"]').type(data.email)
  cy.get('input[name="mobile"]').type(data.mobilenumber)
  cy.root().submit()
 // cy.get('.btn-primary').click()
})
		
		//cy.get(':nth-child(1) > .form-control').type(data.name)
		
		//cy.get(':nth-child(2) > .form-control').type(data.email)
		//cy.get(':nth-child(3) > .form-control').type(data.mobilenumber)
		
      })
   })
})
